creator-node/src/userReqMiddleware.js

```diff
@@ -0,0 +1,59 @@
+const { userReqLimiter } = require('./reqLimiter')
+const { getSPInfo, verifyRequesterIsValidSP } = require('./components/replicaSet/URSMRegistrationComponentService')
+
+/*
+if ((route == 'batch_clock' or route == 'individual clock')) {
+  try {
+    isValidSP()
+    next()
+  } catch (E) {
+    // probably not a valid SP. this is ok
+  }
+}
+rateLimit()
+*/
+
+// For the clock fetching routes, if the requester if from a valid SP, do not enforce
+// a rate limit as we do not want to rate limit SPs checking other SPs for clock values.
+// The clock fetching routes are used in issuing potential reconfig ops.
+async function ensureValidSPRequesterMiddleware (req, res, next) {
```

Comment:
this logic can probably be moved inside the userReqLimiter by adding a skip function like we do in identity https://github.com/AudiusProject/audius-protocol/blob/master/identity-service/src/app.js#L228

Reply:
+1

Reply:
discusssed via slack: looks like the skip field doesn't take async functions, so let's keep it as a separate middleware